File: backend/src/ads/dto/create-ad.dto.ts
import { IsEnum, IsNotEmpty, IsOptional, IsString } from 'class-validator';

export class CreateAdDto {
    @IsString()
    @IsNotEmpty()
    title: string;
  
    @IsString()
    @IsNotEmpty()
    description: string;
  
    @IsString()
    @IsNotEmpty()
    poster_id: string;
  
    @IsString()
    @IsNotEmpty()
    location: string;
  
    @IsString()
    @IsNotEmpty()
    category: string;
  
}

File: backend-dotnet/Soa.Contracts/Dtos/CreateAdDto.cs
using System.Text.Json.Serialization;

namespace Soa.Contracts.Dtos;

public class CreateAdDto
{
    [JsonPropertyName("title")]
    public string Title { get; set; } = null!;

    [JsonPropertyName("description")]
    public string Description { get; set; } = null!;

    [JsonPropertyName("poster_id")]
    public string PosterId { get; set; } = null!;

    [JsonPropertyName("location")]
    public string Location { get; set; } = null!;

    [JsonPropertyName("category")]
    public string Category { get; set; } = null!;
}
